From: NeilBrown <neilb@suse.de>
Subject: Fix connection reuse when sk_error_report is used.
Patch-mainline: Not needed
References: bnc#930972

Now that we are using sk_error_report, we need to clear
sk_err when reusing the connection.

Upstream doesn't need this because they got rid of connection
reuse.  This is a rather big change so best kept out of
SLE12 unless we really need it.

Signed-off-by: NeilBrown <neilb@suse.de>

---
 net/sunrpc/xprtsock.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/net/sunrpc/xprtsock.c
+++ b/net/sunrpc/xprtsock.c
@@ -2143,6 +2143,8 @@ static void xs_abort_connection(struct s
 	 */
 	memset(&any, 0, sizeof(any));
 	any.sa_family = AF_UNSPEC;
+	if (transport->inet)
+		transport->inet->sk_err = 0;
 	result = kernel_connect(transport->sock, &any, sizeof(any), 0);
 	trace_rpc_socket_reset_connection(&transport->xprt,
 			transport->sock, result);
